inst/benchmarking.Rmd

```diff
@@ -0,0 +1,106 @@
+---
+title: "Benchmarking mlr3fda"
+output: html_document
+---
+
+```{r}
+library(data.table)
+library(mlr3fda)
+library(mlr3pipelines)
+
+setDTthreads(threads = 1)
+
+generate_data <- function(n_patients = 100, n_weeks = 10, seed = 1234) {
+  set.seed(seed)
+  lapply(paste0("patient_", seq(1, n_patients)), function(patient_id) {
+    tibble::tibble(
+      patient_id = as.factor(patient_id),
+      week = sample(1:n_weeks, n_weeks, replace = FALSE),
+      systolic_bp = rnorm(n_weeks, 120, 15),
+      diastolic_bp = rnorm(n_weeks, 80, 10),
+      heart_rate = rnorm(n_weeks, 70, 10),
+      age = sample(20:90, n_weeks, replace = TRUE),
+      gender = as.factor(sample(c("Male", "Female"), n_weeks, replace = TRUE)),
+      recovery_rate = systolic_bp * 0.3 + diastolic_bp * 0.4 + heart_rate * 0.2
+    )
+  }) |>
+    dplyr::bind_rows()
+}
+
+analyse_dplyr <- function(patients, window_start, window_end) {
+  patients |>
+    dplyr::filter(between(week, window_start, window_end)) |>
+    dplyr::group_by(patient_id, measurement_type) |>
+    dplyr::summarise(
+      mean = mean(measurement_value),
+      var = var(measurement_value),
+      slope = coef(lm(measurement_value ~ week))[[2]],
+      .groups = "drop"
+    )
+}
+
+analyse_dt <- function(patients, window_start, window_end) {
+  patients <- as.data.table(patients)
+  patients[between(week, window_start, window_end), .(
+    mean = mean(measurement_value),
+    var = var(measurement_value),
+    slope = coef(lm(measurement_value ~ week))[[2]]
+  ), keyby = .(patient_id, measurement_type)]
+}
+
+analyse_fda <- function(task, left = -Inf, right = Inf) {
+  po_fmean <- po("ffs",
```

Comment:
I think the PipeOp construction should be outside of this function. The PipeOp is often called way more often that it is constructed.